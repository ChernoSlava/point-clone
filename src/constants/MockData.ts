export const mockArticles = [
  {
    id: 1,
    title: { short: "Article 1" },
    description: { intro: "Description 1" },
    thumbnail: "thumbnail1.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 2,
    title: { short: "Article 2" },
    description: { intro: "Description 2" },
    thumbnail: "thumbnail2.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 3,
    title: { short: "Article 3" },
    description: { intro: "Description 3" },
    thumbnail: "thumbnail3.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 4,
    title: { short: "Article 4" },
    description: { intro: "Description 4" },
    thumbnail: "thumbnail4.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 5,
    title: { short: "Article 5" },
    description: { intro: "Description 5" },
    thumbnail: "thumbnail5.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 6,
    title: { short: "Article 6" },
    description: { intro: "Description 6" },
    thumbnail: "thumbnail6.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 7,
    title: { short: "Article 7" },
    description: { intro: "Description 7" },
    thumbnail: "thumbnail7.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 8,
    title: { short: "Article 8" },
    description: { intro: "Description 8" },
    thumbnail: "thumbnail8.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 9,
    title: { short: "Article 9" },
    description: { intro: "Description 9" },
    thumbnail: "thumbnail9.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 10,
    title: { short: "Article 10" },
    description: { intro: "Description 10" },
    thumbnail: "thumbnail10.jpg",
    dates: { posted: "1630860959" },
  },
];

export const mockArticlesSmall = [
  {
    id: 1,
    title: { short: "Article 1" },
    description: { intro: "Description 1" },
    thumbnail: "thumbnail1.jpg",
    dates: { posted: "1630860959" },
  },
  {
    id: 2,
    title: { short: "Article 2" },
    description: { intro: "Description 2" },
    thumbnail: "thumbnail2.jpg",
    dates: { posted: "1630860959" },
  },
];
