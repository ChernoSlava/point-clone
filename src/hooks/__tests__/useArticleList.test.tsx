import { MockedProvider } from "@apollo/client/testing";
import { mockArticles } from "@constants/MockData";
import { renderHook } from "@testing-library/react-hooks";
import { ARTICLES_QUERY } from "@utils/index";

import { useArticleList } from "..";

const mocks = [
  {
    request: {
      query: ARTICLES_QUERY,
      variables: { skip: 0 },
    },
    result: {
      data: {
        contents: mockArticles,
      },
    },
  },
];

describe("useArticleList", () => {
  it("should fetch article list", async () => {
    const { unmount, result, waitForNextUpdate } = renderHook(
      () => useArticleList(),
      {
        wrapper: MockedProvider,
        initialProps: { mocks },
      }
    );

    expect(result.current.loading).toBe(true);
    expect(result.current.error).toBeUndefined();
    expect(result.current.articles).toHaveLength(0);
    expect(result.current.loadingMore).toBe(false);

    await waitForNextUpdate();

    expect(result.current.loading).toBe(false);
    expect(result.current.error).toBeUndefined();
    expect(result.current.articles).toEqual(mockArticles);
    expect(result.current.loadingMore).toBe(false);

    expect(result.current).toMatchSnapshot();
    unmount();
  });
});
