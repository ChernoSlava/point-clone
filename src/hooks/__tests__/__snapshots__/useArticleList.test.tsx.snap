// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useArticleList should fetch article list 1`] = `
{
  "articles": [
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 1",
      },
      "id": 1,
      "thumbnail": "thumbnail1.jpg",
      "title": {
        "short": "Article 1",
      },
    },
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 2",
      },
      "id": 2,
      "thumbnail": "thumbnail2.jpg",
      "title": {
        "short": "Article 2",
      },
    },
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 3",
      },
      "id": 3,
      "thumbnail": "thumbnail3.jpg",
      "title": {
        "short": "Article 3",
      },
    },
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 4",
      },
      "id": 4,
      "thumbnail": "thumbnail4.jpg",
      "title": {
        "short": "Article 4",
      },
    },
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 5",
      },
      "id": 5,
      "thumbnail": "thumbnail5.jpg",
      "title": {
        "short": "Article 5",
      },
    },
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 6",
      },
      "id": 6,
      "thumbnail": "thumbnail6.jpg",
      "title": {
        "short": "Article 6",
      },
    },
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 7",
      },
      "id": 7,
      "thumbnail": "thumbnail7.jpg",
      "title": {
        "short": "Article 7",
      },
    },
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 8",
      },
      "id": 8,
      "thumbnail": "thumbnail8.jpg",
      "title": {
        "short": "Article 8",
      },
    },
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 9",
      },
      "id": 9,
      "thumbnail": "thumbnail9.jpg",
      "title": {
        "short": "Article 9",
      },
    },
    {
      "dates": {
        "posted": "1630860959",
      },
      "description": {
        "intro": "Description 10",
      },
      "id": 10,
      "thumbnail": "thumbnail10.jpg",
      "title": {
        "short": "Article 10",
      },
    },
  ],
  "error": undefined,
  "loadMore": [Function],
  "loading": false,
  "loadingMore": false,
}
`;
